R/ptype-abbr-full.R

```diff
@@ -56,11 +58,12 @@ vec_ptype_full.default <- function(x, ...) {
 }
 
 #' @export
-vec_ptype_abbr.default <- function(x, ...) {
+vec_ptype_abbr.default <- function(x, ..., check_named = FALSE, shape = TRUE) {
```

Comment:
Maybe these arguments should be handled in the generic rather than in methods? Otherwise all implementations will need to deal with them.

Alternatively, these could be arguments for the default method only, and we'd use `ellipsis::check_dots_used()` in the generic. This is more straightforward so might make more sense as a first step.

Reply:
I don't see how this can be handled in the generic. The call to `UseMethod()` never returns. We could implement

```r
vec_ptype_label <- function(x, ... type = c("abbr", "full"), prefix_named = FALSE, suffix_shape = TRUE) {
  ...
}
```

and use that in pillar instead, would you support this?

(We now have quite a few packages implementing `vec_ptype_abbr()` methods, so changing `vec_ptype_abbr()` to a non-generic (and implementing a different generic) won't fly.)

Reply:
> The call to UseMethod() never returns.

You can call `UseMethod("vec_ptype_abbr")` from a helper function.

Reply:
Wow, clever! Do you have an example or documentation for this... feature?

Reply:
oh just make sure there is also an unreachable `UseMethod()` call in the generic for static analysis (e.g. roxygen)

```r
out <- vec_ptype_abbr_dispatch(x, ...)

...

return(out)
UseMethod("vec_ptype_abbr")
```

Reply:
Are there any requirements to the signature of the helper function?

Reply:
It should take the arguments dispatched to the methods. Here I'd make it `x, ...` to make it clearer that methods should take `...` (even though I think R CMD check will require these arguments based on the signature of the fake generic).

Reply:
Thanks, I can do this.